components/SearchBar.js

```diff
@@ -1,42 +1,79 @@
-import IconButton from "@material-ui/core/IconButton"
-import InputAdornment from "@material-ui/core/InputAdornment"
-import SearchIcon from "@material-ui/icons/Search"
-import { TextField } from "@material-ui/core"
-import { Autocomplete } from "@material-ui/lab"
+import { useState } from "react"
+import TextField from "@material-ui/core/TextField"
+import Autocomplete from "@material-ui/lab/Autocomplete"
+import { useSearchMovies } from "../hooks/useSearchMovies"
+import CircularProgress from "@material-ui/core/CircularProgress"
+import { debounce } from "throttle-debounce"
+
+const TIME_BETWEEN_NETWORK_REQUESTS = 200
+
+export default function Asynchronous() {
+    const [open, setOpen] = useState(false)
+    const [options, setOptions] = useState([])
+    const [isLoading, setIsLoading] = useState(false)
+    const loading = open && isLoading
+
+    const handleChange = (e) => {
+        setIsLoading(true)
+
+        debounceFunc(e.target.value)
+    }
+
+    const handleClose = () => {
+        debounceFunc.cancel()
+        setOpen(false)
+        setOptions([])
+    }
+
+    const debounceFunc = debounce(
```

Comment:
debouncing network requests so it doesn't spam the api

Reply:
Sweet